cli/azd/pkg/apphost/aca_ingress_test.go

```diff
@@ -0,0 +1,227 @@
+			External:   false,
+			TargetPort: 8080,
+		},
+		Transport:              acaIngressTransportHttp2,
+		AllowInsecure:          true,
+		AdditionalPortMappings: []genContainerAppIngressAdditionalPortMappings(nil),
+	}
+	ingress, err = buildAcaIngress(bindings, 8080)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedIngress, ingress)
+
+	// Test case 3: Multiple external endpoints
+	bindings = map[string]*Binding{
+		"80": {
+			External:  true,
+			Scheme:    acaIngressSchemaHttp,
+			Transport: acaIngressTransportHttp2,
+		},
+		"443": {
+			External:  true,
+			Scheme:    acaIngressSchemaHttps,
+			Transport: acaIngressTransportHttp2,
+		},
+		"8080": {
+			TargetPort: to.Ptr(90),
+			External:   true,
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+	}
+	ingress, err = buildAcaIngress(bindings, 8080)
+	assert.Error(t, err)
+	assert.EqualError(t, err, "Multiple external endpoints are not supported")
+	assert.Nil(t, ingress)
+
+	// Test case 4: Multiple internal only HTTP(s) endpoints
+	bindings = map[string]*Binding{
+		"http": {
+			Scheme:    acaIngressSchemaHttp,
+			Transport: acaIngressTransportHttp2,
+		},
+		"https": {
+			Scheme:    acaIngressSchemaHttps,
+			Transport: acaIngressTransportHttp2,
+		},
+		"other": {
+			TargetPort: to.Ptr(90),
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+	}
+	ingress, err = buildAcaIngress(bindings, 8080)
+	assert.Error(t, err)
+	assert.EqualError(t, err, "Multiple internal only HTTP(s) endpoints are not supported")
+	assert.Nil(t, ingress)
+
+	// Test case 5: More than 5 additional ports
+	bindings = map[string]*Binding{
+		"80": {
+			TargetPort: to.Ptr(80),
+			External:   false,
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+		"443": {
+			TargetPort: to.Ptr(443),
+			External:   false,
+			Scheme:     acaIngressSchemaHttps,
+			Transport:  acaIngressTransportHttp2,
+		},
+		"8080": {
+			TargetPort: to.Ptr(8080),
+			External:   true,
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+		"8081": {
+			TargetPort: to.Ptr(8081),
+			External:   false,
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+		"8082": {
+			TargetPort: to.Ptr(8082),
+			External:   false,
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+		"8083": {
+			TargetPort: to.Ptr(8083),
+			External:   false,
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+		"8084": {
+			TargetPort: to.Ptr(8084),
+			External:   false,
+			Scheme:     acaIngressSchemaHttp,
+			Transport:  acaIngressTransportHttp2,
+		},
+	}
+	ingress, err = buildAcaIngress(bindings, 8080)
+	assert.Error(t, err)
+	assert.EqualError(t, err, "More than 5 additional ports are not supported. "+
+		"See https://learn.microsoft.com/en-us/azure/container-apps/ingress-overview#tcp for more details.")
```

Comment:
Can you remove the `en-us` part of this URL?  Azure will redirect to the correct location based on the user's language (other projects in the larger azure-sdk org have some linters that run over markdown files and warn about this, which is why I know this trivia)